src/infrastructure/arranging/ordering.rs

```diff
@@ -234,18 +243,19 @@ impl<T> Arranging for OrderingStream<T> {
     /// This can only happen in cases where we have a duplicated package. Again we don't give anything back.
     ///
     /// # Remark
-    /// - When we receive an item there is a possibility that a gab is filled and one or more items will could be returned.
+    /// - When we receive an item there is a possibility that a gap is filled and one or more items will could be returned.
     ///   You should use the `iter_mut` instead for reading the items in order.
     ///   However the item given to `arrange` will be returned directly when it matches the `expected_index`.
     fn arrange(
         &mut self,
         incoming_offset: u16,
         item: Self::ArrangingItem,
     ) -> Option<Self::ArrangingItem> {
+        let incoming_offset = incoming_offset as u16;
```

Comment:
This value is already u16 no?

Reply:
Yes, missed that during rebasing from the previous patch.